From cd082e947a37d811f9c8906e57b40bc4cf27daf5 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Mon, 14 Mar 2016 17:46:53 +0800
Subject: [PATCH 0135/1345] a3700: fdt: align kernel dts files to the u-boot
 dts files

commit  d3dc81bd2eab4f90472c852f109e5dc13bc5da4a from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- kernel's armada-3720-db.dts is aligned to u-boot's armada-3700-db.dts,
  only eth0 network interface is enabled and it works in RGMII mode,
  eth1 interface is removed from kernel's armada-3720-db.dts;
- kernel's armada-3720-db-sgmii1.dts is added and aligned to u-boot's
  armada-3700-db-sgmii1.dts; besides armada-3720-db.dts's configuration,
  eth1 network interface is enabled and it works on SERDES SGMII1
  interface, it uses SMI to do PHY polling, meanwhile eth0 still works
  in RGMII mode

Change-Id: Iccdfcfed62da4b47f714b56407e281c631517b6a
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28215
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 .../boot/dts/marvell/armada-3720-db-sgmii1.dts     |   62 ++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-3720-db.dts     |    6 --
 3 files changed, 63 insertions(+), 6 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 4e0e42c..dea6835 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -4,6 +4,7 @@ dtb-$(CONFIG_ARCH_BERLIN) += berlin4ct-stb.dtb
 
 # Mvebu SoC Family
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sgmii1.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-espressobin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-router.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts
new file mode 100644
index 0000000..e74d732
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-sgmii1.dts
@@ -0,0 +1,62 @@
+/*
+ * Device Tree file for Marvell Armada 3720 development board
+ * (DB-88F3720-DDR3)
+ * Copyright (C) 2016 Marvell
+ *
+ * Ken Ma <make@marvell.com>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "armada-3720-db.dts"
+/* This file enables eth1 to work on SERDES SGMII1 interface and
+ * use SMI to do PHY polling, meanwhile eth0 still works on RGMII mode.
+ */
+/ {
+	soc {
+		internal-regs {
+			eth1: ethernet@40000 {
+				phy-mode = "sgmii";
+				phy = <&phy1>;
+				status = "okay";
+			};
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 33d136e..db24060 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -209,9 +209,3 @@
 	phy = <&phy0>;
 	status = "okay";
 };
-
-&eth1 {
-	phy-mode = "sgmii";
-	phy = <&phy1>;
-	status = "okay";
-};
-- 
1.7.9.5

